src/app/(nextjs_migration)/(authenticated)/user/layout.tsx

```diff
@@ -27,6 +29,32 @@ const MainLayout = async (props: Props) => {
     return <SignInButton autoSignIn />;
   }
 
+  const accountId = session?.user?.subscriber?.fxa_uid;
+
+  let userId;
+  if (accountId) {
+    // If the user is logged in, use a UUID based on the user's subscriber ID.
+    // Note: we may want to use the FxA UID here, but we need approval for that first.
+    // No need to store in a cookie - this is derived from subscriber ID.
+    if (!process.env.NIMBUS_UUID_NAMESPACE) {
+      throw new Error("env var NIMBUS_UUID_NAMESPACE not set");
+    }
+    userId = uuidv5(accountId.toString(), process.env.NIMBUS_UUID_NAMESPACE);
+  }
+
+  if (!userId) {
+    throw new Error("No user ID for Nimbus telemetry");
+  }
+
+  // @see https://github.com/mozilla/experimenter/tree/main/cirrus
+  const serverUrl = process.env.SERVER_URL ?? "http://localhost:6060";
```

Comment:
I almost wish there was a config wrapper layer around env vars or something so we wouldn't need to have fallbacks or cast from string to int peppered in the codebase.
```sh
git grep -Pn "localhost:6\d{3,}" -- ":(exclude)*.md" | cat

.env-dist:7:SERVER_URL=http://localhost:6060
.env-dist:10:NEXTAUTH_URL=http://localhost:6060
playwright.config.js:58:    // baseURL: 'http://localhost:6060',
src/app/(nextjs_migration)/(authenticated)/user/layout.tsx:50:  const serverUrl = process.env.SERVER_URL ?? "http://localhost:6060";
src/app/(nextjs_migration)/(guest)/layout.tsx:53:  const serverUrl = process.env.SERVER_URL ?? "http://localhost:6060";
src/e2e/utils/helpers.js:14:  local: 'http://localhost:6060',
```

But not sure if something like [**convict**](https://www.npmjs.com/package/convict) is overkill (or if Next.js has a better built-in solution). But the other benefit of convict is that we could potentially [disallow process.env.* usage](https://eslint.org/docs/latest/rules/no-process-env) via ESLint to try and force everybody to go through the config layer.

Reply:
@pdehaan would you mind putting this on agenda for tech meeting?

Reply:
Added to agenda for next tech meeting